core/rule-engine/rule-engine-core/src/main/java/com/milesight/beaveriot/rule/observe/RuleEngineContextInterceptor.java

```diff
@@ -40,6 +41,7 @@ protected void cacheOutputArguments(NamedNode definition, Exchange exchange) {
         if (definition.getId().startsWith(RuleFlowIdGenerator.FLOW_ID_PREFIX)) {
             String configNodeId = RuleFlowIdGenerator.removeNamespacedId(exchange.getFromRouteId(), definition.getId());
             exchange.setProperty(configNodeId, exchange.getIn().getBody());
+            exchange.setProperty(ExchangeHeaders.EXCHANGE_FLOW_ID, exchange.getFromRouteId());
```

Comment:
I think this should be in `cacheFromArguments`

Reply:
fix